servicenow/stackstate_checks/servicenow/servicenow.py

```diff
@@ -167,73 +199,181 @@ def _process_components(self, components, instance_config):
+                if old_state is None or old_state != change_request.state:
+                    self._create_event_from_change_request(change_request, instance_info)
+                    crs_persisted_state[change_request.number] = change_request.state
+        self.persistent_state.set_state(self.persistent_instance, state)
+
+    def _load_state(self, instance_info):
+        """
+        Persistent State structure:
+        {
+            url: timestamp,
+            url/change_requests: {
+                change_request_number: state
+            }
+        }
+        timestamp is the date/time of the last sys_updated_on from processed CRs
+
+        :param instance_info: current instance check is processing
+        :return: None
+        """
+        # TODO switch to schematics model
+        self.persistent_state = PersistentState()
+        start_dt = datetime.datetime.now() - datetime.timedelta(days=CRS_OLDEST_UPDATE_IN_NUMBER_OF_DAYS)
+        try:
+            self.persistent_state.get_state(self.persistent_instance)
+        except StateReadException:
+            self.log.info('First run! Creating new persistent state.')
+            empty_state = {
+                instance_info.url: start_dt.strftime(TIME_FORMAT),
+                self.cr_persistence_key: {}
+            }
+            self.persistent_state.set_state(self.persistent_instance, data=empty_state)
+
+    def _create_event_from_change_request(self, change_request, instance_info):
+        cmdb_ci = change_request.cmdb_ci
+        identifiers = []
+        external_id = cmdb_ci['link'].split('/').pop()
+        identifiers.append(external_id)
+        identifiers.append(Identifiers.create_host_identifier(cmdb_ci['display_value']))
+        msg_title = '{}: {}'.format(change_request.number, change_request.short_description)
+        if change_request.description:
+            msg_text = change_request.description
+        else:
+            msg_text = change_request.short_description
+        tags = [
+            'number:{}'.format(change_request.number),
+            'priority:{}'.format(change_request.priority),
+            'risk:{}'.format(change_request.risk),
+            'state:{}'.format(change_request.state),
+            'category:{}'.format(change_request.category),
+            'conflict_status:{}'.format(change_request.conflict_status),
+            'assigned_to:{}'.format(change_request.assigned_to)
+        ]
+        self.event({
+            'timestamp': datetime.datetime.timestamp(change_request.sys_updated_on),
+            'event_type': change_request.type,
+            'msg_title': msg_title,
+            'msg_text': msg_text,
+            # TODO do we need an aggregation_key?
+            # 'aggregation_key': '???',
+            'context': {
+                'source': 'servicenow',
+                'category': 'change_request',
+                'element_identifiers': identifiers,
+                'data': {
+                    'cmdb_ci': cmdb_ci,
```

Comment:
it's a dictionary right ?

Reply:
yes, it is a dictionary. I really don't know if we need to pass `cmdb_ci`. 
We have `identifiers` with sys_id that is used as `external_id` and URN created from cmdb_ci `name`.